src/subxt_transaction.rs

```diff
@@ -496,10 +505,107 @@ pub(crate) fn build_eth_tx_payload(
 	}
 }
 
+#[allow(clippy::too_many_arguments)]
+async fn create_online_transaction<C: subxt::Config, KP, G>(
+	from_keypair: &KP,
+	nonce: u128,
+	mortality: &Option<u64>,
+	account: &str,
+	sink: &SubxtTransactionsSink<C, KP>,
+	from_account_id: &<C as subxt::Config>::AccountId,
+	to_account_id: &<C as subxt::Config>::AccountId,
+	recipe: &TransactionRecipe,
+	generate_payload: G,
+) -> Result<SubxtTransaction<C>, Error>
+where
+	G: GenerateTxPayloadFunction<AccountIdOf<C>>,
+	AccountIdOf<C>: Send + Sync + AsRef<[u8]>,
+	KP: Signer<C> + Clone + Send + Sync + 'static,
+	<<C as subxt::Config>::ExtrinsicParams as subxt::config::ExtrinsicParams<C>>::Params: From<(
+		(),
+		(),
+		(),
+		CheckNonceParams,
+		(),
+		CheckMortalityParams<C>,
+		ChargeAssetTxPaymentParams<C>,
+		ChargeTransactionPaymentParams,
+		(),
+	)>,
+{
+	// Needed because `Params` as associated type does not implement clone, and we need to
+	// recreate the tx params in a loop when we can't create a partial tx with the online
+	// client, due to various RPC related issues or state not being up to date (currently we
+	// handle an error which happens when trying to create a partial tx that is based on a
+	// certain finalized block returned by the RPC, which is then reported as not found).
+	// Retrying seems to fix the issue.
```

Comment:
Is this comment relevant here? Shouldn't be move to the loop itself?
Also not sure if the tx_params function is really needed anymore.